### PR TITLE
Port fallback_formatter_get_possible_formatters() to D8

### PR DESCRIPTION
**Pull request by tonnosf**
Moved method  fallback_formatter_get_possible_formatters() to FieldFormatter class.
Added a check of applicability of each possible formatter.

https://www.drupal.org/node/2626362

### CODE REVIEW

File: fallback_formatter.module [hunk 1]
```diff
@@ -31,39 +31,6 @@ function fallback_formatter_field_formatter_info_alter(&$info) {
...
-    if ($formatter == 'fallback') {
-      continue;
-    }
-    // Check that the field type is allowed for the formatter.
-    elseif (!in_array($field_type, $info['field_types'])) {
-      continue;
-    }
-    // Formatters tagged as 'multiple' that render all items as one element
-    // instead of one element per item delta will fail fallback detection.
-    elseif (isset($info['behaviors']['multiple'])) {
```
Comment (slashrsm) [inline fallback_formatter.module:57]: This condition was lost during the transition it seems.

**Comment by slashrsm** [comment]
Please link to the d.o. issue.

**Comment by Berdir** [comment]
Yes, that flag doesn't exist anymore, there are no longer two difference API's, they just override/implement different methods on the base class. The implementation already works with all of them, we no longer need to check that.

**Comment by slashrsm** [comment]
OK. Merged with a reworded commit message.

**Pull request closed without merging**